heron/tmaster/src/cpp/manager/stateful-restorer.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2015 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "manager/stateful-restorer.h"
+#include <string>
+#include "manager/stmgrstate.h"
+
+namespace heron {
+namespace tmaster {
+
+StatefulRestorer::StatefulRestorer()
+  : in_progress_(false),
+    restore_txid_(0) {
+}
+
+StatefulRestorer::~StatefulRestorer() { }
+
+void StatefulRestorer::StartRestore(const std::string& _checkpoint_id, const StMgrMap& _stmgrs) {
+  if (in_progress_) {
+    LOG(WARNING) << "Starting a Restore for checkpoint "
```

Comment:
Can you clarify what happens in the log message (e.g., do we ignore the request or start another one?). If we're able to start and do another one, why is it a warning and how does shared state of this class in_progress_, restore_txid_, etc) not collide?